Dockerfile

```diff
@@ -40,14 +40,20 @@ RUN set -x; \
     && git submodule update --init skins \
     && git submodule update --init vendor \
     && cd extensions \
-    # VisualEditor
+    # Extensions
     # TODO: make submodules shallow clones?
-    && git submodule update --init VisualEditor \
-    && cd VisualEditor \
-    && git checkout $MEDIAWIKI_VERSION \
-    && git submodule update --init \
-    && cd .. \
-    && git submodule update --init Math
+    && git submodule update --init --recursive VisualEditor \
+    && git submodule update --init --recursive Math \
+    && git submodule update --init --recursive EventBus \
+    && git submodule update --init --recursive Scribunto \
+    && git submodule update --init --recursive ParserFunctions \
+    && git submodule update --init --recursive SyntaxHighlight_GeSHi \
+    && git submodule update --init --recursive Cite \
+    && git submodule update --init --recursive Echo \
+    && git submodule update --init --recursive Flow \
+    && git submodule update --init --recursive PageImages \
+    && git submodule update --init --recursive TextExtracts
```

Comment:
We also need MobileFrontend for the `mobileview` action API used by mobileapps.

Also, don't we need to checkout the specific MW version for each of them?